go/vt/vtgate/planner.go

```diff
@@ -43,10 +52,53 @@ func NewPlanner(vschema *vindexes.VSchema, cacheSize int) *Planner {
 	return plr
 }
 
+// LoadVSchema loads the VSchema from the topo. The function does
+// not return an error. It instead logs warnings on failure.
+func (plr *Planner) LoadVSchema() {
+	formal := &vindexes.VSchemaFormal{
+		Keyspaces: make(map[string]vindexes.KeyspaceFormal),
+	}
+	keyspaces, err := plr.serv.GetSrvKeyspaceNames(context.TODO(), plr.cell)
```

Comment:
Yes, I would pass in the context (that the watcher will need anyway, and can be used to interrupt the watch).

I was planning on doing the watch per keyspace (and not global as you have it there), as an extra interface on the TopoServer, similar to WatchSrvKeyspace in go/vt/topo/server.go. We might need to wrestle with the code a bit to do that.

(these changes can and probably should be a different PR though, we can do that later, proceed with this).